### ISSUE TITLE
CrossHair Tool : Cannot access 'center' before initialization

### ISSUE THREAD

**Issue by Celian-abd** [cornerstonejs/cornerstone3D-beta#79]
Hello, I've been trying to use the crosshair tool while following the exemple available but I could'nt rotate the views.
When I try to rotate on any axes and move the mouse this error occur :
![Screenshot_1](https://user-images.githubusercontent.com/101793092/163971033-9cf353e8-6fa9-4e96-81a6-7d8309184e51.png)
Otherwise all of the others features such as slabThickness are working.

Maybe you got and idea for this problem.

You can see an overview of my crosshair tool :
![Screenshot_2](https://user-images.githubusercontent.com/101793092/163973114-b9bed83e-86ce-4c7d-825e-6ab5f3e8cf89.png)

**Comment by sedghi** [comment]
I tried that example and it worked locally now
Based on your error it is complaining that the init on crosshairs has not been called, which is called on active, passive, enabled modes. Maybe put a debugger at init and debug line by line to see if the toolCenter is calculated

**Comment by Celian-abd** [comment]
When I log my crosshairtool object in the console I can see the toolCenter with coordinates that are the same as my image center coordinates.

**Comment by Celian-abd** [comment]
So here you have the main code where I create the toolgroup and add it all the tools : https://github.com/Pixilib/Dat_Scan_Viewer/blob/main/datscan/src/components/ViewerVolumeRoot.js#L100

And here is my component where I activate the CrossHair : https://github.com/Pixilib/Dat_Scan_Viewer/blob/main/datscan/src/components/tools/coordsOnCursor.js#L247

**Comment by sedghi** [comment]
you should do the toolInit before adding tools https://github.com/Pixilib/Dat_Scan_Viewer/blob/main/datscan/src/components/ViewerVolumeRoot.js#L107

**Comment by Celian-abd** [comment]
Yes this was only a test, but initially the Toolinit was before adding tools and the error was the same.

**Comment by Celian-abd** [comment]
I've updated my code
https://github.com/Pixilib/Dat_Scan_Viewer/blob/main/datscan/src/components/ViewerVolumeRoot.js#L99

**Comment by sedghi** [comment]
what is the recipe for running the app?

**Comment by salimkanoun** [comment]
Hi Alireza, it is a simple Create-React-App,
from the repository go to "datscan" folder and then npm install and npm start.

in the opening webpage you will have a drop zone to drop you dicom files to load them in the volume viewer.

thanks for you help,

Salim

**Comment by sedghi** [comment]
@Celian-abd Is there any specific reason you are using the /dist/esm for the imports? The bundler should automatically use that and you don't need to directly import from each folder

**Comment by Celian-abd** [comment]
I've already tried to `setToolEnable` before `setToolActive` but the result is the same.

**Comment by Celian-abd** [comment]
Solve with lastest version.
